files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -650,4 +680,69 @@ protected function close_handler() {
 	protected function trim_path( $path ) {
 		return ltrim( $path, 'vip:/\\' );
 	}
+
+	/**
+	 * Validates the provided stream arguments for fopen.
+	*
+	* @since   1.0.0
+	* @access  private
+	* @param   string    $path   Path to file
+	* @param   string    $mode   fopen mode
+	*
+	* @return  bool
+	 */
+	private function validate( $path, $mode ) {
+		if ( ! in_array( $mode, [ 'r', 'w', 'a', 'x' ], true ) ) {
```

Comment:
Will `$mode` _always_ have a value?

Reply:
PHP supports more modes than this:

https://www.php.net/manual/en/function.fopen.php

Specifically `r+`, `w+`, `a+`, `x+`, `c`, `c+`, and `e` (not really used). We don't necessarily need to implement those all right away, but we should at least document what is and isn't supported (and probably move the supported modes to a constant).

Reply:
> Will `$mode` _always_ have a value?

Yes as `$mode` is from `fopen()` and is a required argument.

Reply:
> We don't necessarily need to implement those all right away, but we should at least document what is and isn't supported (and probably move the supported modes to a constant).

This is a good idea. I'll add that in. Also, we're ignoring the `+` modes as in our current implementation, all the file handles are read/write capable.

Reply:
Changes made in a145ce99ed9c3643e9f5db25e440a0894d283007

Reply:
Awesome!